internal/sqldb/sqlquery/variable_date.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sqlquery
+
+import (
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/datacommonsorg/mixer/internal/util"
+)
+
+// DateEntityCount returns number of entities (from given candidates) with data for
+// each observation date.
+func DateEntityCount(
+	sqlClient *sql.DB,
+	variables []string,
+	entities []string,
+) (map[string]map[string]map[string]int, error) {
```

Comment:
Wondering if we can use map of structs vs a 4 level map of raw types if it helps readability / future maintenance? Something like this:

```
struct SVDate {
  sv string
  date string
}

struct ProvEntityCount {
  prov string
  entityCount int
}
```

And return `map[SVDate][ProvCount]`.

Reply:
I tried using struct before and found it not improving much because <sv, date1> <sv, date2> becomes two keys, so when iterating the struct, always need to reference back to sv, instead of not working on everything keyed by sv in one loop.